servers/src/grin/sync.rs

```diff
@@ -451,6 +458,11 @@ impl SyncInfo {
 			false
 		}
 	}
+
```

Comment:
Perhaps we could get rid of `fast_sync_due` now. The time seems irrelevant after this fix, so we should be able to only rely on the sync state (including the error), right?

Reply:
We keep `fast_sync_due` for the moment, use it for download timeout detection. Perhaps switch to better timeout detection method in the future.